### PR TITLE
tools/{text/typesetting}: replace name with pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
anybody an idea, what is causing the massive rebuilds here?

**Comment by davidak** [comment]
>anybody an idea, what is causing the massive rebuilds here?

maybe psutils, but not sure

could you just rebase to use staging or would that cause merge conflicts when staging is merged back?

**Comment by Stunkymonkey** [comment]
maybe we can merge `psutils` separately #132398.

**Comment by Stunkymonkey** [comment]
@ofborg eval

File: pkgs/tools/text/catdocx/default.nix [hunk 1]
```diff
@@ -1,7 +1,8 @@
 { stdenv, lib, fetchFromGitHub, makeWrapper, unzip, catdoc }:
 
 stdenv.mkDerivation {
-  name = "catdocx-20170102";
+  pname = "catdocx";
+  version = "2017-01-02";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/catdocx/default.nix:5]:
```suggestion
  version = "unstable-2017-01-02";
```

File: pkgs/tools/text/mpage/default.nix [hunk 2]
```diff
@@ -1,9 +1,11 @@
 { fetchurl, lib, stdenv }:
 
 stdenv.mkDerivation rec {
-  name = "mpage-2.5.7";
+  pname = "mpage";
+  version = "2.5.7";
+
   src = fetchurl {
-    url = "http://www.mesa.nl/pub/mpage/${name}.tgz";
+    url = "http://www.mesa.nl/pub/mpage/mpage-${version}.tgz";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/mpage/default.nix:8]:
```suggestion
    url = "https://www.mesa.nl/pub/mpage/mpage-${version}.tgz";
```

File: pkgs/tools/text/yaml-merge/default.nix [hunk 3]
```diff
@@ -1,7 +1,8 @@
 { lib, stdenv, fetchFromGitHub, pythonPackages }:
 
 stdenv.mkDerivation {
-  name= "yaml-merge-2016-02-16";
+  pname = "yaml-merge";
+  version = "2016-02-16";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/yaml-merge/default.nix:5]:
```suggestion
  version = "unstable-2016-02-16";
```

File: pkgs/tools/typesetting/hevea/default.nix [hunk 4]
```diff
@@ -1,10 +1,11 @@
 { lib, stdenv, fetchurl, ocamlPackages }:
 
 stdenv.mkDerivation rec {
-  name = "hevea-2.35";
+  pname = "hevea";
+  version = "2.35";
 
   src = fetchurl {
-    url = "http://pauillac.inria.fr/~maranget/hevea/distri/${name}.tar.gz";
+    url = "http://pauillac.inria.fr/~maranget/hevea/distri/hevea-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/typesetting/hevea/default.nix:8]:
```suggestion
    url = "https://pauillac.inria.fr/~maranget/hevea/distri/hevea-${version}.tar.gz";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
also please target staging

**Comment by jonringer** [comment]
for rebasing, you can do:
```
common=$(git merge-base origin/master origin/staging)
git rebase --onto=$common HEAD~4 # or however many commits you want to carry over
git push ... ... --force
```
then change the base branch to the relevant branch.

**Comment by Stunkymonkey** [comment]
I forgot to change the base in the PR itself.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 131709` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>diffstat</li>
  </ul>
</details>

**Comment by davidak** [comment]
Building this would lead to download more than 50 GB of sources. I don't have that much free space and i don't think it's useful.

So should we just merge this after reviewing the code changes?

I have tested one package to see if it generally evaluates.

**Review by davidak — APPROVED** [review]
Changes look good.

**Pull request merged**